### ISSUE TITLE
Authenticator UI does not respond to submit

### ISSUE THREAD

**Issue by zirkelc** [aws-amplify/amplify-ui#1476]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

CRA

### Please describe your bug.

My App is using the `Authenticator` component from `@aws-amplify/ui-react`. I noticed yesterday that the Form for Login or Signup does not respond to the submit button. It checks if all mandatory fields are filled, but then it does not continue with Login/Signup. It does nothing and there are not errors on the console.

### What's the expected behaviour?

Handle the click action and continue with Login or Signup.

### Help us reproduce the bug!

I tried to reproduce the issue in a separate repository, but it did work as expected. I would be glad for any tips what this issue might be.

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

package.json:
- "@aws-amplify/ui": "^3.2.0",
- "@aws-amplify/ui-react": "^2.8.0",
- "aws-amplify": "^4.3.15",

**Comment by keevcodes** [comment]
also experiencing this issue

**Comment by ErikCH** [comment]
Hi @keevcodes and @zirkelc ?
Are either one of you using xstate?

We have a compatibility issue with XState that's listed [here](https://github.com/aws-amplify/amplify-ui/issues/1226) . Since we are using it internally, if you install your own version of XState, it duplicates it and causes issues with the buttons not working for logging in or signing up.

A quick work around for now while we wait on an update for Xstate is to add this to your package.json

```
"resolutions": {
    "xstate": "4.28.0"
},
```
Use `overrides` instead of `resolutions` if you're using npm.

**Comment by ErikCH** [comment]
Also, we have a working version of CRA running every 15 minutes in our canary that's running the latest. It runs a sign in and sign up. You can check out this version [here](https://github.com/aws-amplify/amplify-ui/tree/main/canary/apps/react/cra).

Perhaps if the issue isn't in XState, maybe there is something else that you are missing? 

Could you provide a list of what's in your `package.json` ?

**Comment by zirkelc** [comment]
@ErikCH pinning `xstate` to 4.28.0 did not solve the issue. I verified that only the Amplify packages depend on it:

```
zirkelc@Chris-MacBook-Pro studio % npm ls xstate
@nebula/studio@1.0.0 /Users/zirkelc/Code/nebula/packages/studio
├─┬ @aws-amplify/ui-react@2.8.0
│ └─┬ @xstate/react@1.6.3
│   └── xstate@4.28.1
└─┬ @aws-amplify/ui@3.2.0
  └── xstate@4.28.0
```

However, downgrading to `"@aws-amplify/ui-react": "2.2.0"` did work.

**Comment by ErikCH** [comment]
hi @zirkelc !

Could you provide a snippet of your `package.json`? That would help determine if there are other libraries that might be conflicting.

**Comment by keevcodes** [comment]
@ErikCH thanks for info and help. Indeed this seemed like an xstate dependency issue as the amplify-cli was also using version 4.30.4 of xstate. Removing the amplify-cli dependency did the trick.
![Screenshot 2022-03-07 at 10 18 22](https://user-images.githubusercontent.com/17259420/157038290-d2fe4aa1-711c-4b94-9ab9-30be901823d8.png)

**Comment by zirkelc** [comment]
@ErikCH of course, here are the dependencies of my frontend project:

````
  "dependencies": {
    "@aws-amplify/ui-react": "2.2.0",
    "@emotion/react": "^11.4.1",
    "@emotion/styled": "^11.3.0",
    "@hookform/devtools": "^4.0.1",
    "@hookform/error-message": "^2.0.0",
    "@hookform/resolvers": "^2.8.8",
    "@mui/icons-material": "^5.3.1",
    "@mui/lab": "^5.0.0-alpha.67",
    "@mui/material": "^5.4.0",
    "@mui/styles": "^5.3.0",
    "@mui/x-data-grid-pro": "^5.4.0",
    "@testing-library/jest-dom": "^5.11.9",
    "@testing-library/react": "^11.2.3",
    "@testing-library/user-event": "^12.6.0",
    "@types/jest": "^26.0.20",
    "@types/node": "^12.19.13",
    "@types/react": "^17.0.30",
    "@types/react-dom": "^17.0.9",
    "autosuggest-highlight": "^3.1.1",
    "aws-amplify": "^4.3.15",
    "axios": "^0.21.1",
    "babel-plugin-prismjs": "^2.0.1",
    "clsx": "^1.1.1",
    "cytoscape": "^3.18.0",
    "cytoscape-cola": "^2.4.0",
    "cytoscape-cose-bilkent": "^4.1.0",
    "cytoscape-cxtmenu": "^3.4.0",
    "cytoscape-edgehandles": "^4.0.1",
    "date-fns": "^2.19.0",
    "graphql": "^16.2.0",
    "graphql-request": "^3.7.0",
    "jsonata": "^1.8.4",
    "libphonenumber-js": "^1.9.49",
    "lodash": "^4.17.20",
    "nanoid": "^3.1.20",
    "notistack": "^2.0.3",
    "prismjs": "^1.23.0",
    "querystring": "^0.2.0",
    "react": "^17.0.1",
    "react-color": "^2.19.3",
    "react-copy-to-clipboard": "^5.0.3",
    "react-cytoscapejs": "^1.2.1",
    "react-dom": "^17.0.1",
    "react-hook-form": "^7.26.1",
    "react-masonry-css": "^1.0.16",
    "react-qr-code": "^2.0.3",
    "react-query": "^3.15.2",
    "react-router-dom": "^5.2.0",
    "react-scripts": "^4.0.3",
    "react-simple-code-editor": "^0.11.0",
    "react-split-pane": "^0.1.92",
    "typescript": "^4.3.5",
    "use-debounce": "^6.0.1",
    "use-neo4j": "^0.3.0",
    "web-vitals": "^0.2.4",
    "yup": "^0.32.11"
  },
```

**Comment by zirkelc** [comment]
> @ErikCH thanks for info and help. Indeed this seemed like an xstate dependency issue as the amplify-cli was also using version 4.30.4 of xstate. Removing the amplify-cli dependency did the trick. ![Screenshot 2022-03-07 at 10 18 22](https://user-images.githubusercontent.com/17259420/157038290-d2fe4aa1-711c-4b94-9ab9-30be901823d8.png)

Interesting, the dependency of `@aws-amplify/cli@7.6.22` to `xstate@4.30.3` didn't appear in my output, because I installed the Amplify CLI globally:

![2022-03-07_14-43-55](https://user-images.githubusercontent.com/950244/157045722-71a873c9-07d1-4047-94c2-c147f2191678.png)

Maybe the solution via `overrides` / `resolutions` didn't work because of the Amplify CLI global installation.

**Comment by keevcodes** [comment]
@zirkelc could be. I initially tried the overrides solution to no avail

**Comment by ErikCH** [comment]
Glad to hear @zirkelc ! I'm not sure why the resolutions/overrides did not work for you though. That should lock the dependency tree to that version.

@keevcodes Did you try using resolutions or overrides? Are you also using amplify-cli?

**Comment by wlee221** [comment]
Hi all, we just updated our xstate dependency to be more compatible to avoid package duplications.

This will be released next Tuesday, and will let you know once that happens. This should remove the need to use `overrides` or `resolutions`.
